src/Persistence_matrix/include/gudhi/persistence_interval.h

```diff
@@ -0,0 +1,215 @@
+/*    This file is part of the Gudhi Library - https://gudhi.inria.fr/ - which is released under MIT.
```

Comment:
File should be named `Persistence_interval.h`, cf. [code convention](https://github.com/GUDHI/gudhi-devel/blob/master/.github/code_conventions.md#c)

Reply:
It is not the only file which doesn't follow this convention in the module. Is it fine if I correct them all in this PR or should I do another one? Also the `example_*` are not respected.

Reply:
Please don't mix mass renaming and actual changes in the same branch, it makes it super hard to review.

Reply:
I agree, but as the changes of this PR are mostly in just one file, I thought there could be an exception. But no problem in doing another PR for the renamings.

Reply:
> as the changes of this PR are mostly in just one file

If you rename things all over the place in other files, it will take a reviewer a long time just to make sure that this file is the only one with non-trivial changes.
(I didn't look at what the renaming was about)

Reply:
True, but I don't think there will be other reviewesr than you two, no? But, like I said, it makes sense to make another PR out of it, it was more of a "you never know" question. The new PR is here: #1106.

Reply:
@VincentRouvreau I am not sure anymore if I should really rename `persistence_interval.h` to `Persistence_interval`. The convention says:
```
- If a single class or function is provided in a file, its name (with the same letter case) should be used for the file name.
- If a file does not contain a single class, its name should not begin with a capital letter.
```
The file does not contain only the `Persistence_interval` class, but also different global methods (the get etc.) and global structures (tuple_element). So, the second point of the convention should apply, no?

Reply:
(I never read the convention and don't really care)
I would argue that those are part of the interface of this class. Whether they are free functions, static members, etc is a detail.

Reply:
I agree, that is why they are in the same file. But I am not sure that is how I interpret the convention. I saw the two namings a way for the developer to search more easily: "if there is a upper case, don't need to look here for this one method related to it, it has to be somewhere else". Otherwise, I am not sure to see the utility to have two different naming convention for files...